AutoRest/Generators/Azure.Common/Azure.Common/AzureCodeGenerator.cs

```diff
@@ -332,12 +332,16 @@ public static bool IsDerivedFromExternalResource(CompositeType compositeType)
                 return false;
             }
 
-            return compositeType.BaseModelType != null &&
-                   (compositeType.BaseModelType.Name.Equals(ResourceType, StringComparison.OrdinalIgnoreCase) ||
-                    compositeType.BaseModelType.Name.Equals(SubResourceType, StringComparison.OrdinalIgnoreCase)) &&
-                   compositeType.BaseModelType.Extensions.ContainsKey(ExternalExtension) &&
-                   compositeType.BaseModelType.Extensions[ExternalExtension] is bool &&
-                   (bool)compositeType.BaseModelType.Extensions[ExternalExtension];
+            if (compositeType.BaseModelType != null &&
+                (compositeType.BaseModelType.Name.Equals(ResourceType, StringComparison.OrdinalIgnoreCase) ||
+                 compositeType.BaseModelType.Name.Equals(SubResourceType, StringComparison.OrdinalIgnoreCase)) &&
+                compositeType.BaseModelType.Extensions.ContainsKey(ExternalExtension))
```

Comment:
The intention is to accept:

x-ms-external

-or

x-ms-external: true

but if 

x-ms-external: false is set, then this would be disabled